ext/oss/sink.go

```diff
@@ -60,41 +56,25 @@ func NewSink(commonSink common.Sink,
 	}
 
 	o := &OSSSink{
-		Sink:                      commonSink,
-		client:                    client,
-		destinationURITemplate:    tmpl,
-		writeHandlers:             make(map[string]xio.WriteFlusher),
-		ossHandlers:               make(map[string]io.WriteCloser),
-		fileRecordCounters:        make(map[string]int),
-		partialFileRecordCounters: make(map[string]int),
-		batchSize:                 batchSize,
-		enableOverwrite:           enableOverwrite,
-		skipHeader:                skipHeader,
-		maxTempFileRecordNumber:   maxTempFileRecordNumber,
+		Sink:                    commonSink,
+		client:                  client,
+		destinationURITemplate:  tmpl,
+		writeHandlers:           make(map[string]xio.WriteFlusher),
```

Comment:
let's create one interface, `xio.WriteFlushCloser` and use this instead to close the writer